BaseApp/TAB99008535.TXT

```diff
@@ -0,0 +1,207 @@
+    Time=13:35:23;
+    Modified=Yes;
+    Version List=NAVW111.00;
+  }
+  PROPERTIES
+  {
+    CaptionML=[ENU=TempBlob;
+               ISL=TempBlob];
+  }
+  FIELDS
+  {
+    { 1   ;   ;Primary Key         ;Integer       ;CaptionML=[ENU=Primary Key;
+                                                              ISL=A�allykill] }
+    { 2   ;   ;Blob                ;BLOB          ;CaptionML=[ENU=Blob;
+                                                              ISL=BLOB] }
+  }
+  KEYS
+  {
+    {    ;Primary Key                             ;Clustered=Yes }
+  }
+  FIELDGROUPS
+  {
+  }
+  CODE
+  {
+    VAR
+      GlobalInStream@1000 : InStream;
+      GlobalOutStream@1012 : OutStream;
+      ReadLinesInitialized@1001 : Boolean;
+      NoContentErr@1002 : TextConst 'ENU=The BLOB field is empty.;ISL=The BLOB field is empty.';
+      UnknownImageTypeErr@1003 : TextConst 'ENU=Unknown image type.;ISL=Unknown image type.';
+      XmlCannotBeLoadedErr@1004 : TextConst 'ENU=The XML cannot be loaded.;ISL=The XML cannot be loaded.';
+      WriteLinesInitialized@1013 : Boolean;
+      WritingTextLinesNotInitializedErr@1014 : TextConst 'ENU=Please use start writing text lines fuction before adding text';
+
+    [External]
+    PROCEDURE WriteAsText@3(Content@1001 : Text;Encoding@1000 : TextEncoding);
+    VAR
+      OutStr@1002 : OutStream;
+    BEGIN
+      CLEAR(Blob);
+      IF Content = '' THEN
+        EXIT;
+      Blob.CREATEOUTSTREAM(OutStr,Encoding);
+      OutStr.WRITETEXT(Content);
+    END;
+
+    [External]
+    PROCEDURE ReadAsText@5(LineSeparator@1001 : Text;Encoding@1000 : TextEncoding) Content : Text;
+    VAR
+      InStream@1002 : InStream;
+      ContentLine@1003 : Text;
+    BEGIN
+      Blob.CREATEINSTREAM(InStream,Encoding);
+
+      InStream.READTEXT(Content);
+      WHILE NOT InStream.EOS DO BEGIN
+        InStream.READTEXT(ContentLine);
+        Content += LineSeparator + ContentLine;
+      END;
+    END;
+
+    [External]
+    PROCEDURE StartReadingTextLines@8(Encoding@1000 : TextEncoding);
+    BEGIN
+      Blob.CREATEINSTREAM(GlobalInStream,Encoding);
+      ReadLinesInitialized := TRUE;
+    END;
+
+    PROCEDURE StartWritingTextLines@13(Encoding@1000 : TextEncoding);
+    BEGIN
+      CLEAR(Blob);
+      Blob.CREATEOUTSTREAM(GlobalOutStream,Encoding);
+      WriteLinesInitialized := TRUE;
+    END;
+
+    [External]
+    PROCEDURE MoreTextLines@9() : Boolean;
+    BEGIN
+      IF NOT ReadLinesInitialized THEN
+        StartReadingTextLines(TEXTENCODING::Windows);
+      EXIT(NOT GlobalInStream.EOS);
+    END;
+
+    [External]
+    PROCEDURE ReadTextLine@7() : Text;
+    VAR
+      ContentLine@1003 : Text;
+    BEGIN
+      IF NOT MoreTextLines THEN
+        EXIT('');
+      GlobalInStream.READTEXT(ContentLine);
+      EXIT(ContentLine);
+    END;
+
+    PROCEDURE WriteTextLine@14(Content@10034985 : Text);
+    BEGIN
+      IF NOT WriteLinesInitialized THEN
+        ERROR(WritingTextLinesNotInitializedErr);

```

Comment:
I prefer to initialize directly. We do that for read as well. Will default to same encoding as when writing. We should not use errors, which are for developers, if we can avoid it. Errors will surface to the user, who will not know what to do about it.